backend/src/main/java/com/bakdata/conquery/models/jobs/ImportJob.java

```diff
@@ -249,21 +248,19 @@ private static Map<String, DictionaryMapping> importDictionaries(DistributedName
 					  ResourceUtil.throwNotFoundIfNull(dictionaryId, sharedDictionary);
 					  log.trace("Merging into shared Dictionary[{}]", sharedDictionary);
 
-					  mapping = DictionaryMapping.createAndImport(importDictionary, sharedDictionary);
-				  }
+					  final DictionaryMapping mapping = DictionaryMapping.createAndImport(importDictionary, sharedDictionary);
+
+					  if (mapping.getNumberOfNewIds() != 0) {
+						  storeAndDistributeDictionary(namespace, mapping.getTargetDictionary());
```

Comment:
während dem speichern muss das lock auch gehalten werden, damit es nichtt während dem schreiben bearbeitet wird. Königsklasse wäre natürlich nur einmal zu mergen und zu speichern, das sehe ich aber nicht im scope von einem hotfix

Reply:
kontext ist, dass hier ein sharedDictionary in einer Tabelle mehrfach verwendet wird und es so zu concurrent Modification kommt.